x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -364,21 +364,21 @@ public String toString() {
 
     }
 
-    static class AuditableRestRequest extends AuditableRequest {
+    static class AuditableHttpRequest extends AuditableRequest {
 
-        private final RestRequest request;
+        final BasicHttpRequest request;
         private final String requestId;
 
-        AuditableRestRequest(
+        AuditableHttpRequest(
             AuditTrail auditTrail,
             AuthenticationFailureHandler failureHandler,
             ThreadContext threadContext,
-            RestRequest request
+            BasicHttpRequest request
         ) {
             super(auditTrail, failureHandler, threadContext);
             this.request = request;
-            // There should never be an existing audit-id when processing a rest request.
-            this.requestId = AuditUtil.generateRequestId(threadContext);
+            // There might be an existing audit-id (e.g. generated by the rest request) but there might not be (e.g. an internal action)
+            this.requestId = AuditUtil.getOrGenerateRequestId(threadContext);
```

Comment:
Using `AuditUtil#getOrGenerateRequestId` instead of `AuditUtil#generateRequestId` is just a temporary measure.  Ultimately `HttpRequest`s will only be intercepted once so we can fall back to the stricter generate.